docs/src/modules/components/MarkdownDocsFooter.js

```diff
@@ -0,0 +1,352 @@
+
+  if (description === undefined) {
+    throw new Error('Missing description in the page');
+  }
+
+  const setCurrentRatingFromCookie = React.useCallback(() => {
+    setCurrentRating(getCurrentRating(currentPage.pathname));
+  }, [currentPage.pathname]);
+
+  React.useEffect(() => {
+    setCurrentRatingFromCookie();
+  }, [setCurrentRatingFromCookie]);
+
+  async function processFeedback(rating, comment) {
+    const result = await submitFeedback(currentPage.pathname, rating, comment);
+    if (result) {
+      setSnackbarMessage(t('feedbackSubmitted'));
+    } else {
+      setCurrentRatingFromCookie();
+      setSnackbarMessage(t('feedbackFailed'));
+    }
+    setSnackbarOpen(true);
+  }
+
+  const handleClickUp = async () => {
+    setCurrentRating(1);
+    await processFeedback(1);
+  };
+
+  const handleClickDown = () => {
+    setCurrentRating(0);
+    setCommentOpen(true);
+  };
+
+  const handleCloseComment = async (comment) => {
+    setCommentOpen(false);
+    if (comment !== null) {
+      await processFeedback(0, comment);
+    } else {
+      setCurrentRatingFromCookie();
+    }
+  };
+
+  const handleCloseSnackbar = () => {
+    setSnackbarOpen(false);
+  };
+
+  return (
+    <React.Fragment>
+      <Comment open={commentOpen} onClose={handleCloseComment} />
+      <footer className={classes.footer}>
+        {!currentPage ||
+        currentPage.displayNav === false ||
+        (nextPage.displayNav === false && !prevPage) ? null : (
+          <React.Fragment>
+            <Divider />
+            <div className={classes.pagination}>
+              {prevPage ? (
+                <Button
+                  component={Link}
+                  naked
+                  href={prevPage.pathname}
+                  size="large"
+                  className={classes.pageLinkButton}
+                  startIcon={<ChevronLeftIcon />}
+                >
+                  {pageToTitleI18n(prevPage, t)}
+                </Button>
+              ) : (
+                <div />
+              )}
+              <div role="group" aria-label={t('feedbackGroupLabel')}>
+                <Typography
+                  variant="subtitle1"
+                  component="span"
+                  className={classes.feedbackMessage}
+                >
+                  {t('ratingMessage')}
+                </Typography>
+                <Tooltip title={t('feedbackYes')}>
+                  <IconButton onClick={handleClickUp} aria-pressed={currentRating === 1}>
+                    <ThumbUpIcon color={currentRating === 1 ? 'primary' : undefined} />
+                  </IconButton>
+                </Tooltip>
+                <Tooltip title={t('feedbackNo')}>
+                  <IconButton onClick={handleClickDown} aria-pressed={currentRating === 0}>
+                    <ThumbDownIcon color={currentRating === 0 ? 'error' : undefined} />
+                  </IconButton>
+                </Tooltip>
```

Comment:
I guess ideally we want to use `ToggleButton` here but can't because of current limitations of `ToggleButtonGroup` (similar to https://github.com/mui-org/material-ui/issues/18091 and https://github.com/mui-org/material-ui/issues/14346)? 

Might be a good time to switch to context because this seems like a pretty common use case. Though it's not blocking this PR.